src/pages/workspaces/workspace/JobHistory.js

```diff
@@ -119,7 +121,23 @@ const JobHistory = _.flow(
 
     try {
       this.setState({ loading: true })
-      const submissions = _.orderBy('submissionDate', 'desc', await Workspaces.workspace(namespace, name).listSubmissions())
+      const submissions = _.flow(
+        _.orderBy('submissionDate', 'desc'),
+        _.map(sub => {
```

Comment:
Before I merge this, I'd like to know if you think this is a terrible idea, @panentheos

Reply:
Assuming there were performance issues with transforming this on the fly, this seems like a reasonable optimization. We could also `memoize`, but I think that's largely equivalent.